test/system/src/main/java/io/pravega/test/system/framework/DockerRemoteSequential.java

```diff
@@ -0,0 +1,162 @@
+import com.spotify.docker.client.DefaultDockerClient;
+import com.spotify.docker.client.DockerClient;
+import com.spotify.docker.client.exceptions.DockerException;
+import com.spotify.docker.client.messages.Container;
+import com.spotify.docker.client.messages.ContainerConfig;
+import com.spotify.docker.client.messages.ContainerCreation;
+import com.spotify.docker.client.messages.ContainerInfo;
+import com.spotify.docker.client.messages.HostConfig;
+import com.spotify.docker.client.messages.PortBinding;
+import io.pravega.common.concurrent.FutureHelpers;
+import lombok.extern.slf4j.Slf4j;
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import static org.junit.Assert.assertFalse;
+
+@Slf4j
+public class DockerRemoteSequential implements TestExecutor {
+
+    public final static DockerClient CLIENT = DefaultDockerClient.builder().uri("http://"+ System.getProperty("masterIP")+ ":2375").build();
+    private final static String IMAGE = System.getProperty("dockerImageRegistry")+ "/java:8";
+    public String id;
+    private final ScheduledExecutorService executorService = Executors.newScheduledThreadPool(3);
+
+    public CompletableFuture<Void> startTestExecution(Method testMethod) {
+
+        log.debug("Starting test execution for method: {}", testMethod);
+
+        String className = testMethod.getDeclaringClass().getName();
+        String methodName = testMethod.getName();
+        String containerName = methodName + ".testJob";
+
+        return CompletableFuture.runAsync(() -> {
+            startTest(containerName, className, methodName);
+        }).thenCompose(v2 -> waitForJobCompletion(containerName))
+                .whenComplete((v, ex) -> {
+            stopContainer(containerName); //stop container once execution is complete.
+                    if (ex != null) {
+                        log.error("Error while executing the test. ClassName: {}, MethodName: {}", className,
+                                methodName);
+                    }
+                }
+                );
+    }
+
+
+    @Override
+    public CompletableFuture<Void> stopTestExecution(String testID) {
+        throw new org.apache.commons.lang.NotImplementedException("Stop Execution is not used for Remote sequential execution");
+    }
+
+    private CompletableFuture<Void> waitForJobCompletion(final String containerName) {
+        return FutureHelpers.loop(() -> isTestRunning(containerName),
+                () -> FutureHelpers.delayedFuture(Duration.ofSeconds(3), executorService),
+                executorService);
+    }
+
+    private boolean isTestRunning(String containerName) {
```

Comment:
We can reuse the containerID returned by startContainer.

Reply:
done.